src/mudlet-lua/tests/UI_spec.lua

```diff
@@ -81,4 +81,63 @@ describe("Tests UI functions", function()
       assert.are.equal(outputhtml, copy2html("testconsole"))
     end)
   end)
+
+  describe("Test the operation of the windowType() function", function()
+    it("Should identify an existing label correctly", function()
+      createLabel("testlabel", 0,0,0,0, 1)
+
+      assert.are.equal(windowType("testlabel"), "label")
+    end)
+
+    it("Should not identify a non-existing label", function()
+      assert.are.equal(windowType("fake label"), nil)
+    end)
+
+    it("Should identify the main window as 'main'", function()
+      assert.are.equal(windowType("main"), "main")
+    end)
+
+    it("Should identify an existing userwindow correctly", function()
+      openUserWindow("testuserwindow")
+
+      assert.are.equal(windowType("testuserwindow"), "userwindow")
+    end)
+
+    it("Should not identify a non-existing userwindow", function()
```

Comment:
This is redundant - if it doesn't exist then it **can't** have a *type* - so only one of the non-existing tests is required.